### PR TITLE
fix(generative-ui): report the depth warning at the nesting level it enforces

### LINKED ISSUE #5784: react-generative-ui: data_table caps are half Slack's limit, and four converter warnings misreport what happened
the 2026-08-10 converter audit turned up four verified discrepancies across the `./slack` and `./teams` subpaths. none of them throw. they either produce output smaller than the platform accepts, or attach a warning that describes something other than what the converter actually did. filing them together because they are one cleanup pass, not four features.

### 1. `data_table` caps are half Slack's current limit

`DATA_TABLE_ROW_CAP = 100` and `DATA_TABLE_CHAR_BUDGET = 10000` (`packages/react-generative-ui/src/slack/constants.ts:115,124`) come from the July reading of the platform docs. the [current data table reference](https://docs.slack.dev/reference/block-kit/blocks/data-table-block/) documents:

> You can have a minimum of 2 rows (1 regular row plus the header) and a maximum of 201 rows (200 regular rows plus the header).

> A single table's character count across all cells cannot exceed 20,000 characters. Additionally, the aggregate character count across all table cells for a single message cannot exceed 20,000 characters.

so a 150 row table is clamped to 100 with a `clamped` warning even though Slack would have accepted all of it. `DATA_TABLE_COLUMN_CAP = 20` is still correct.

fix: 100 to 200 rows, 10000 to 20000 characters, and update the two cap tests.

### 2. Teams warnings quote 100 KB but fire at 80,000 bytes

`PAYLOAD_SOFT_CAP = 80000` (`packages/react-generative-ui/src/teams/constants.ts:70`), but both emit sites report the platform ceiling instead of the budget that was actually crossed:

- `teams/toAdaptiveCard.ts:721` "the card is ${size} bytes, exceeding Teams' 100 KB bot message limit."
- `teams/toTeamsAttachments.ts:118` "the carousel attachments total ${size} bytes, exceeding Teams' 100 KB bot message limit."

a payload between 80,000 and 102,400 bytes is told it exceeded a limit it did not exceed.

fix: name the 80,000 byte soft budget as the thing that was crossed, and keep 100 KB as the stated reason for the headroom.

### 3. three Teams `clamped` warnings clamp nothing

- `Row` with more than 3 children warns `clamped` "Teams recommends at most 3 columns" and then maps every child into a column anyway (`teams/toAdaptiveCard.ts:542-549`)
- buttons past `PRIMARY_ACTION_CAP` warn `clamped` with detail "actions beyond 6 were set to secondary mode" (`teams/toAdaptiveCard.ts:211-223`), which is a style change, not a removal
- a renamed input id warns `clamped` (`teams/toAdaptiveCard.ts:108-122`), which is a rename

`clamped` is the code a consumer reads to decide whether content was lost, and in all three cases nothing was. `SlackConversionWarning` / the Teams twin are `"clamped" | "dropped" | "fallback"`, so this needs a call: reclassify these three as `fallback`, or widen the union with an advisory code. widening is not free, it breaks an exhaustive switch on the consumer side even though the type itself is only being added to.

### 4. the depth warning says 64 levels, the real ceiling is 32

`MAX_TRAVERSAL_DEPTH = 64` (`slack/constants.ts:15`, documented as "the maximum normalized traversal depth") is spent 2 units per nesting level, because `boundNode` charges one for the element and one for its `children` array (`slack/toSlackBlocks.ts:1205,1225`). measured against the built dist with a chain of nested `Card`s:

```
levels=32 depthWarning=no
levels=33 depthWarning=YES
```

the emitted detail still reads "nodes deeper than 64 levels were dropped". the two existing tests nest 65 and 70 levels (`slack/toSlackBlocks.test.ts:1736,1880`), both well past the real boundary, so nothing pins where it actually is.

fix: report the effective element depth in the warning, and add a test at the 32/33 boundary. the Teams `boundSpec` is kept line for line parallel with this walk, so it needs the same check in the same pass.

### scope

all four are in `packages/react-generative-ui`, which is published, so this needs a patch changeset. `./slack` and `./teams` have no in-repo code consumers yet, so the blast radius is the published converter output and its warnings.

<a href="https://rupic.dev/s/s2.1lVIGNc_rQGTUALidu2hMsGmKWmOH-zAHElT73FjRCTl3hhGDikZpvFgGwo?ref=github" target="_blank" rel="noopener noreferrer"><picture><source media="(prefers-color-scheme: dark)" srcset="https://rupic.dev/badges/track-dark-v1.svg">Track in Rupic</picture></a>

### PR DESCRIPTION
**Pull request by okisdev**
closes #5784

## summary

- `MAX_TRAVERSAL_DEPTH = 64` is spent two units per level of element nesting, because the bounded pre-pass charges one for the element and one for its `children` array (`slack/toSlackBlocks.ts:1205,1225`, `teams/boundSpec.ts:45,65`), so the effective ceiling is 32 levels while the warning claimed 64
- measured against the built dist, both converters first drop at **33** levels of nesting, not 65:

  ```
  levels=32 slack="none"                                     teams="none"
  levels=33 slack="nodes deeper than 32 levels were dropped." teams="nodes deeper than 32 levels were dropped."
  ```

- both details now use a derived `MAX_ELEMENT_DEPTH`, and the constant's JSDoc says what unit it counts in, since "the maximum normalized traversal depth" is what made 64 read as tree levels in the first place
- the two existing tests nest 70 levels, far past the boundary, so nothing pinned where it actually is; each converter gains a test asserting no drop at 32 and a drop at 33
- the docs said "a depth ceiling of 64" on the Teams page and left it unnamed on the Slack page; both now say 32 levels of element nesting

the ceiling itself is unchanged. making 64 units mean 64 levels would double how deep a model-supplied tree may nest, and that bound exists precisely because the tree arrives from a model.

### already verified

- [x] `pnpm -C packages/react-generative-ui test` -> 548/548 green
- [x] `pnpm lint` -> exit 0
- [x] `pnpm api-surface` and `pnpm -C apps/docs generate:api-reference` -> no drift, the constants are internal
- [x] the 32/33 boundary measured against the built dist for both converters, output above

### reviewer should verify

- [ ] a 33-level `Card` chain warns `nodes deeper than 32 levels were dropped.` and a 32-level chain converts with no depth warning

## notes

- last of the four fixes from #5784, so this closes it. the earlier three are #5785, #5787, and #5788
- two related defects surfaced while reviewing #5788 and are tracked separately: #5790 (Teams paths that discard content with no warning at all) and #5793 (the Slack `degradeCard` site, which warns once for two different outcomes and truncates silently)

<a href="https://cubic.dev/pr/assistant-ui/assistant-ui/pull/5794?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

<a href="https://rupic.dev/s/s2.KgBKyI6O57RxHx8bFkbcjUQ08j2b2XZIvR2PADrm5J1TPTiq7Nvs82A9CBY?ref=github" target="_blank" rel="noopener noreferrer"><picture><source media="(prefers-color-scheme: dark)" srcset="https://rupic.dev/badges/track-dark-v1.svg">Track in Rupic</picture></a>

### CODE REVIEW

File: packages/react-generative-ui/src/teams/constants.ts [hunk 1]
```diff
@@ -17,9 +17,16 @@ export const ADAPTIVE_CARD_SCHEMA =
...
-/** The maximum normalized traversal depth. */
+/**
+ * The maximum traversal depth of the bounded pre-pass, counted in visited
+ * values rather than tree levels. An element and its `children` array are two
+ * values, so a level of element nesting spends two.
+ */
 export const MAX_TRAVERSAL_DEPTH = 64;
 
+/** The element-nesting ceiling {@link MAX_TRAVERSAL_DEPTH} works out to. */
+export const MAX_ELEMENT_DEPTH = MAX_TRAVERSAL_DEPTH / 2;
```
[the remark this answers is not in the document]
Comment (okisdev) [reply packages/react-generative-ui/src/teams/constants.ts]: both fixed. the halving is now attributed to `boundNode` rather than to the constant, and the boundary tests carry the array-root shape alongside the object-root one, measured: an object root drops at 33 wrappers, an array root at 32. one correction on the supporting detail, since it changes whether the two Slack walks agree: Slack's `convertSequence` also spends two per level (`toSlackBlocks.ts:1158` descends into `convertElement` at `depth + 1`, which then recurses at `depth + 1` again), so it matches its own `boundNode`. it is Teams where the two conventions differ, exactly as `toAdaptiveCard.ts:625-628` documents. the conclusion stands either way: the rule is not a property of the shared constant.
